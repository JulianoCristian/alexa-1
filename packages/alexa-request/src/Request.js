import { Request as Constants } from 'alexa-constants';

export default class Request {
  static session = (...args) => new Request().session(...args);
  static intent = (...args) => new Request().intent(...args);
  static launchRequest = (...args) => new Request().launchRequest(...args);
  static sessionEndedRequest = (...args) => new Request().sessionEndedRequest(...args);

  constructor(state = {}) {
    this.state = state;
  }

  session(session) {
    return new Request({
      ...this.state,
      session: {
        ...this.state.session,
        ...session
      }
    });
  }

  intent(name, slots = {}) {
    const slotData = Object.entries(slots).reduce((state, [name, value]) => ({ ...state, [name]: { name, value } }), {});
    return new Request({
      ...this.state,
      request: {
        type: Constants.Intent,
        intent: {
          name: name,
          ...(Object.keys(slotData).length ? { slots: slotData } : null)
        }
      }
    });
  }

  launchRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.Launch
      }
    });
  }

  sessionEndedRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.SessionEnded
      }
    });
  }

  build() {
    return { ...this.state };
  }
}
