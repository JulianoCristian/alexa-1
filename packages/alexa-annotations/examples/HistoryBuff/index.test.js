import Request from 'alexa-request';
import HistoryBuff from './index';

test('launch', () => {
  const event = Request.launchRequest().build();
  return expect(HistoryBuff(event)).resolves.toMatchSnapshot();
});

test.skip('GetFirstEventIntent', () => {
  const event = Request.intent('GetFirstEventIntent', { day: 465302909000 });
  return expect(HistoryBuff(event)).resolves.toMatchSnapshot();
});
