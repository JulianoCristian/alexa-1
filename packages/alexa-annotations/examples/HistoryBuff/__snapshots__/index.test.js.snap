// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`launch 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "History Buff. What day do you want events for?",
      "title": "This Day in History",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "ssml": "<speak><p>History buff.</p><p>What day do you want events for?</p></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "With History Buff, you can get historical events for any day of the year. For example, you could say today, or August thirtieth. Now, which day do you want?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;
